win32/winutil.c

```diff
@@ -36,6 +36,13 @@ PHP_WINUTIL_API char *php_win32_error_to_msg(HRESULT error)
 		return "";
 	}
 
+	size_t lenw = wcslen(bufw);
+	if (lenw > 0) {
+		/* strip trailing line breaks and periods */
+		for (pw = bufw + lenw - 1; *pw == L'\r' || *pw == L'\n' || *pw == L'.'; pw--);
```

Comment:
This looks a bit unsafe. If the string only contains newlines and `.`, it will read past the end. Check `pw >= bufw`?

Reply:
Ah, right, thanks!